src/interpreter/Node.h

```diff
@@ -401,12 +402,27 @@ class ConditionalOperation {
 };
 
 /**
- * @class Constant
+ * @class NumericConstant
  */
-class Constant : public Node {
+class NumericConstant : public Node {
     using Node::Node;
 };
 
+/**
+ * @class StringConstant
+ */
+class StringConstant : public Node {
+public:
+    StringConstant(enum NodeType ty, const ram::Node* sdw, size_t constant)
```

Comment:
Is `enum` necessary?

```suggestion
    StringConstant(enum NodeType ty, const ram::Node* sdw, std::size_t constant)
```

Also, is `sdw` optional?  Could it be a `&`?

Reply:
Yes - it is necessary because we use a switch statement for dispatch.

Reply:
I mean "I don't think you need to say`enum NodeType`, just `NodeType`".  I updated my suggestion.

Reply:
We need to further refine/refactor the interpreter. Xiaowen is on this job.